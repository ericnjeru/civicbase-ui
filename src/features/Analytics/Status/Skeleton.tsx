import * as Stats from 'components/Stats'
import tw from 'twin.macro'

const Skeleton = () => {
  return (
    <div css={tw`mt-24`}>
      <Stats.List>
        <Stats.Skeleton />
        <Stats.Skeleton />
        <Stats.Skeleton />
        <Stats.Skeleton />
      </Stats.List>
    </div>
  )
}

export default Skeleton
