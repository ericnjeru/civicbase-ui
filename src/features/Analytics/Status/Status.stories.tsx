import { ComponentMeta, ComponentStory } from '@storybook/react'
import { AnalyticsProvider } from 'contexts/analytics'
import { survey } from 'test/sample'
import Status from './Status'

export default {
  title: 'Features/Analytics/Status',
  component: Status,
} as ComponentMeta<typeof Status>

const Template: ComponentStory<typeof Status> = () => {
  return (
    <div style={{ width: 900 }}>
      <AnalyticsProvider survey={survey}>
        <Status />
      </AnalyticsProvider>
    </div>
  )
}

export const Basic = Template.bind({})

Basic.args = {}
