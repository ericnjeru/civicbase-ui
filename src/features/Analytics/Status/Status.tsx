import * as Stats from 'components/Stats'
import { useAnalytics } from 'contexts/analytics'
import tw from 'twin.macro'
import { toCamelCase } from 'utilities/util'

import Skeleton from './Skeleton'

const Status = () => {
  const { isLoading, survey } = useAnalytics()

  if (!survey || isLoading) {
    return <Skeleton />
  }

  return (
    <div css={tw`mt-24`}>
      <Stats.List>
        <Stats.Item title="Total Respondents" metric={0}>
          {/* <Badge style={{ height: 'min-content' }} css={tw`bg-green-200 flex items-center text-green-900`}>
            <AiOutlineArrowUp css={tw`mr-1 text-green-600`} />
            {respondentsIncrement}%
          </Badge> */}
        </Stats.Item>
        <Stats.Item title="Total Access" metric={0}>
          {/* <Badge style={{ height: 'min-content' }} css={tw`bg-green-200 flex items-center text-green-900`}>
            <AiOutlineArrowUp css={tw`mr-1 text-green-600`} />
            {accessIncrement}%
          </Badge> */}
        </Stats.Item>
        <Stats.Item title="Convertion Rate" metric={`${0}%`}>
          {/* {convertionIncrement && convertionIncrement > 0 ? (
            <Badge
              style={{ height: 'min-content' }}
              css={[
                tw`flex items-center`,
                convertionIncrement && convertionIncrement > 0
                  ? tw`bg-green-200 text-green-900`
                  : tw`bg-red-200 text-red-900`,
              ]}
            >
              {convertionIncrement && convertionIncrement > 0 ? (
                <AiOutlineArrowUp css={tw`mr-1 text-green-600`} />
              ) : (
                <AiOutlineArrowDown css={tw`mr-1 text-red-600`} />
              )}
              {convertionIncrement}%
            </Badge>
          ) : null} */}
        </Stats.Item>
        <Stats.Item title="Current Status" metric={toCamelCase(survey.status)} />
      </Stats.List>
    </div>
  )
}

export default Status
