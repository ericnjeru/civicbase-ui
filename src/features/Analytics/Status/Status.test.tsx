import { render } from '@testing-library/react'
import { survey } from 'test/sample'

import Status from './Status'

describe('Analytics Status', () => {
  const customComponent = ({ ...props } = {}) => {
    return render(<Status survey={{ ...survey, ...props }} />)
  }

  it('should show access', () => {
    const { container } = customComponent()

    expect(container).toMatchSnapshot()
  })
})
