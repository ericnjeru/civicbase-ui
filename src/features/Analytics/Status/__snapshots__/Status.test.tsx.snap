// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Status should show access 1`] = `
<div>
  <div
    class="Skeleton___StyledDiv-sc-vmp43t-0 fOItRj"
  >
    <div
      class="sc-bdnxRM Stats___StyledCard-sc-1bbp4h3-0 YHPbv irAMoJ"
    >
      <div
        class="Stats___StyledDiv-sc-1bbp4h3-1 bRvies"
      >
        <div
          class="Skeleton___StyledDiv-sc-lwldx-4 jDHENG"
        >
          <div
            class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine-sc-lwldx-3 bhtXAR fYewdA"
          />
          <div
            class="Skeleton___StyledDiv2-sc-lwldx-5 fJpFce"
          >
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine2-sc-lwldx-2 bhtXAR bkepcT"
            />
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine3-sc-lwldx-1 bhtXAR bWmReg"
            />
          </div>
        </div>
        <div
          class="Skeleton___StyledDiv-sc-lwldx-4 jDHENG"
        >
          <div
            class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine-sc-lwldx-3 bhtXAR fYewdA"
          />
          <div
            class="Skeleton___StyledDiv2-sc-lwldx-5 fJpFce"
          >
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine2-sc-lwldx-2 bhtXAR bkepcT"
            />
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine3-sc-lwldx-1 bhtXAR bWmReg"
            />
          </div>
        </div>
        <div
          class="Skeleton___StyledDiv-sc-lwldx-4 jDHENG"
        >
          <div
            class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine-sc-lwldx-3 bhtXAR fYewdA"
          />
          <div
            class="Skeleton___StyledDiv2-sc-lwldx-5 fJpFce"
          >
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine2-sc-lwldx-2 bhtXAR bkepcT"
            />
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine3-sc-lwldx-1 bhtXAR bWmReg"
            />
          </div>
        </div>
        <div
          class="Skeleton___StyledDiv-sc-lwldx-4 jDHENG"
        >
          <div
            class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine-sc-lwldx-3 bhtXAR fYewdA"
          />
          <div
            class="Skeleton___StyledDiv2-sc-lwldx-5 fJpFce"
          >
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine2-sc-lwldx-2 bhtXAR bkepcT"
            />
            <div
              class="Skeleton__TextLine-sc-lwldx-0 Skeleton___StyledTextLine3-sc-lwldx-1 bhtXAR bWmReg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
